src/main/java/com/ruralsourcing/WeatherBot/service/SlackIntegration.java

```diff
@@ -0,0 +1,99 @@
+package com.ruralsourcing.WeatherBot.service;
+
+import com.github.seratch.jslack.Slack;
+import com.github.seratch.jslack.api.model.event.MessageEvent;
+import com.github.seratch.jslack.api.rtm.RTMClient;
+import com.github.seratch.jslack.api.rtm.message.Message;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.ruralsourcing.WeatherBot.model.ResponseModel;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+
+/**
+ *
+ */
+public class SlackIntegration implements Runnable {
+    private static final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+    private static final String TOKEN = System.getenv("SLACK_BOT_API_TOKEN");
+
+    private final AtomicReference<ResponseModel> reference;
+
+    public SlackIntegration(AtomicReference<ResponseModel> reference) {
+        this.reference = reference;
+    }
+
+    @Override
+    public void run() {
+        rtmMessage();
+    }
+
+    private void rtmMessage() {
+        final Slack slack = Slack.getInstance();
+
+        try (final RTMClient rtm = slack.rtm(TOKEN)) {
+            rtm.addMessageHandler((message) -> {
+                final ResponseModel currentData = reference.get();
+
+                JsonObject json = JsonParser.parseString(message).getAsJsonObject();
+
+                if (json.get("type") == null) {
+                    return;
+                }
+
+                final String type = json.get("type").getAsString();
+
+                if(!"message".equals(type)){
+                    return;
+                }
+
+                final MessageEvent event = gson.fromJson(json, MessageEvent.class);
+
+                //do stuff in response:
+                System.out.println(event);
+
+                //If the message came from a DM or WeatherBot was pinged using @, respond
+                if(event.getText().contains("@UPXMYNW93") || event.getChannel().startsWith("D")){
+                    rtm.sendMessage(Message.builder()
+                            .id(System.currentTimeMillis())
+                            .channel(event.getChannel())
+                            .text(currentDataString(currentData))
+                            .build().toJSONString());
+                }
+            });
+
+
+            rtm.connect();
+
+            while (true) {
+                Thread.onSpinWait();
```

Comment:
I unintentionally lied to you on this, `Thread.onSpinWait();` should only be used for short waiting on conditions. Instead, do a standard sleep loop here:

```java
while (true) {
    try {
        Thread.sleep(500);
    } catch (final InterruptedException e) {
        break;
    }
}
```